### ISSUE TITLE
routing needs to check dest VPC

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#121]
TLDR: Need to add VPC awareness to the routing layer to enforce VPC separation.

* [x] Write unit test attempting to send cross VPC and verify traffic doesn't pass. This should fail.
* [x] Implement VPC enforcement in routing layer.
* [x] Verify unit test passes.

While testing changes for #118 I found that a ping between two guests on the same host was failing (from the perspective of the guest performing the ping). However, when looking at the packet capture for the other guest (the ping destination) I noticed both the ICMP Echo arriving as well as an ICMP Echo Reply being returned -- odd. Furthermore, in the system log was the following message which indicates the packet was looped-back but the destination was not found:

```
WARNING: underlay dest is same as src but the Port was not found vni = 10, mac = A8:40:25:FF:00:02
```

It then hit me that the likely problem was I actually had the guests on two different VPCs (represented by the VNI); and sure enough that was the case. But if that's the case how did the ICMP Echo ever make it to the destination? Well turns out the router layer currently has no notion of VPC, and thus we always send the packet to the destination, regardless of VPC, so long as there is at least a router entry for it. This is not correct. Our routing layer needs to consider VPC/VNI, and only if there exists a "VPC Peering Gateway" entry should inter-VPC communication be allowed (discussed in RFC 21 §2.9).

You might wonder, if the Echo made it across VNI boundaries, how come the Reply didn't also make it across? Turns out this was caused by inconsistent data between how I created the xde devices and what I set in the global Virtual-2-Physical mapping, effectively allowing only one way comms.
